src/core/geometry/qgsregularpolygon.h

```diff
@@ -0,0 +1,213 @@
+     * \param center The center of the regular polygon.
+     * \param radius Distance from the center and the first vertice or sides (see \a circle).
+     * \param azimuth Angle in degrees started from the North to the first vertice.
+     * \param numSides Number of sides of the regular polygon.
+     * \param circle Option to create the polygon: inscribed in circle (the radius is the distance between the center and vertices) or circumscribed about circle (the radius is the distance from the center to the midpoints of the sides).
+     */
+    QgsRegularPolygon( const QgsPointV2 center, const double radius, const double azimuth, const int numSides, const int circle );
+
+    /** Constructs a regular polygon by center and another point.
+     * \param center The center of the regular polygon.
+     * \param pt1 The first vertice if the polygon is inscribed in circle or the midpoint of a side if the polygon is circumscribed about circle.
+     * \param numSides Number of sides of the regular polygon.
+     * \param circle Option to create the polygon inscribed in circle (the radius is the distance between the center and vertices) or circumscribed about circle (the radius is the distance from the center to the midpoints of the sides).
+     */
+    QgsRegularPolygon( const QgsPointV2 center, const QgsPointV2 pt1, const int  numSides, const int circle );
+
+    /** Constructs a regular polygon by two points of the first side.
+     * \param pt1 The first vertice of the first side, also first vertice of the regular polygon.
+     * \param pt2 The second vertice of the first side.
+     * \param numSides Number of sides of the regular polygon.
+     */
+    QgsRegularPolygon( const QgsPointV2 pt1, const QgsPointV2 pt2, const int  numSides );
+
+    virtual bool operator ==( const QgsRegularPolygon &rp ) const;
+    virtual bool operator !=( const QgsRegularPolygon &rp ) const;
+
+    //! A regular polygon is empty if radius equal to 0 or number of sides < 3
+    virtual bool isEmpty() const;
+
+    /** Returns the center point of the regular polygon.
+     * \see setCenter()
+     */
+    QgsPointV2 center() const {return mCenter; }
+
+    /** Returns the radius.
+     * This is also the radius of the circumscribing circle.
+     * \see apothem()
+     * \see setRadius()
+     */
+    double radius() const {return mRadius; }
+
+    /** Returns the first vertice (corner) of the regular polygon.
+     * \see setVertice()
+     */
+    QgsPointV2 vertice() const {return mVertice; }
+
+    /** Returns the apothem of the regular polygon.
+     * The apothem is the radius of the inscribed circle.
+     * \see radius()
+     */
+    double apothem() const {return mRadius * cos( M_PI / mNumSides ); }
+
+    /** Returns the number of sides of the regular polygon.
+     * \see setNumSides()
+     */
+    int numSides() const {return mNumSides; }
```

Comment:
can you change this to either sides() or numberSides()? We are removing abbreviations from the API in 3.0 to keep things consistent and predictable